zebra-consensus/src/chain/tests.rs

```diff
@@ -250,15 +254,7 @@ async fn verify_fail_add_block_checkpoint() -> Result<(), Report> {
     Ok(())
 }
 
-/*
-// This test is disabled because it doesn't test the right thing:
-// the BlockVerifier and CheckpointVerifier make different requests
-// and produce different transcripts.
-
-
 #[tokio::test]
-// Temporarily ignore this test, until the state can handle out-of-order blocks
-#[ignore]
 async fn continuous_blockchain_test() -> Result<(), Report> {
```

Comment:
This test on the other hand, I'm not sure if this one is doable anymore, given that we have a hard requirement on only chain verifying post sapling blocks. This test only ever uses blocks one thru ten, all of which are only valid in a checkpointing context. Not sure how to move forward on this one.

Reply:
Yeah, testing contextual verification is going to be hard - and we didn't really cover testing in the State RFC.

I can see the following options:
* create a test mode in the block verifier that skips checks which only work after Sapling activation
* create a test mode in the state service that starts the chain at Sapling
* only test contextual verification in long-running node deployment tests
* do nothing

If we don't want test modes, we can create test services - but we would need to keep them up-to-date with the non-test versions.

Maybe we should do a small design RFC for semantic and contextual verification tests?

Reply:
It would be cool if we didn't have hardcoded testnet and mainnet configs and instead had support for arbitrarily configured networks, then we could just test on arbitrary new block chains we invent that only ever use v4 transactions and whose finalized block set only includes genesis.

Reply:
We can add a `regression test` network config - zcashd has one.

Reply:
We don't have a completely working state implementation yet, because `CommitBlock` is not yet implemented:
https://github.com/ZcashFoundation/zebra/blob/86ed13060f15a96877e8315c3e50d8dfb57561eb/zebra-state/src/service.rs#L55

So I suggest that we leave this test disabled (and unmodified) for now, open a ticket, and try to re-enable it after we have implemented `CommitBlock`.